### PR TITLE
fix: remove deduplication logic in pipeline for input

### PR DESCRIPTION
**Pull request by Vic-Dev**
#### What problem is being solved?
When handling deduplication in List Objects, we need to consider:
1. Input to recursive relations
2. Input to tuple cycles
3. Output when not in exclusion or intersection (exclusion and intersection already have post-processing of output that should handle duplicates)

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved deduplication logic for item processing, addressing edge cases with recursive and cyclic relationships.
  * Optimized list operations to use the pipeline-based processing path for non-streamed requests.

* **Tests**
  * Added comprehensive test coverage for deduplication scenarios to ensure correct handling of duplicate items in complex operations.

### CODE REVIEW

File: pkg/server/commands/reverseexpand/pipeline.go [hunk 1]
```diff
@@ -411,37 +399,15 @@ func (r *baseResolver) process(ndx int, snd *sender, listeners []*listener) loop
...
-			r.mutexes[ndx].Lock()
-			if _, ok := r.inBuffers[ndx][item.Value]; ok {
-				r.mutexes[ndx].Unlock()
-				continue
-			}
-			r.inBuffers[ndx][item.Value] = struct{}{}
-			r.mutexes[ndx].Unlock()
-			unseen = append(unseen, item)
-		}
+		inItems = append(inItems, msg.Value.Items...)
```
Comment (senojj) [inline pkg/server/commands/reverseexpand/pipeline.go]: this operation seems unnecessary because it simply moves elements from one slice into another.

**Comment by poovamraj** [comment]
@claude

File: pkg/server/commands/reverseexpand/pipeline_test.go [hunk 2]
```diff
@@ -2136,3 +2138,220 @@ func TestPipeline(t *testing.T) {
...
+	cancelFunc func()
+	closeFunc  func()
+}
+
+func (m *mockConsumer) send(g group) {
+	m.sendFunc(g)
+}
+
+func (m *mockConsumer) cancel() {
+	m.cancelFunc()
```
[the remark this answers is not in the document]
Comment (Vic-Dev) [reply pkg/server/commands/reverseexpand/pipeline_test.go:2399]: not needed for these mocks

File: pkg/server/commands/reverseexpand/pipeline_test.go [hunk 3]
```diff
@@ -2136,3 +2138,220 @@ func TestPipeline(t *testing.T) {
...
+	tests := []struct {
+		name           string
+		edge           *Edge
+		inputItems     []Item
+		expectedUnseen []Item
+		expectedOutput []Item
+	}{
+		{
+			name: "recursive_edge_deduplicates",
+			edge: orgMemberEdges[0],
```
[the remark this answers is not in the document]
Comment (Vic-Dev) [reply pkg/server/commands/reverseexpand/pipeline_test.go:2199]: let me live my life Claude

[the remark this answers is not in the document]
Comment (poovamraj) [reply pkg/server/commands/reverseexpand/pipeline_test.go:2199]: You are absolutely right!

**Review by senojj — APPROVED** [review]

File: pkg/server/commands/reverseexpand/pipeline.go [hunk 4]
```diff
@@ -411,31 +411,36 @@ func (r *baseResolver) process(ndx int, snd *sender, listeners []*listener) loop
...
-			if _, ok := r.inBuffers[ndx][item.Value]; ok {
-				r.mutexes[ndx].Unlock()
-				continue
 			}
-			r.inBuffers[ndx][item.Value] = struct{}{}
-			r.mutexes[ndx].Unlock()
-			unseen = append(unseen, item)
+		} else {
+			unseen = msg.Value.Items
 		}
```
[the remark this answers is not in the document]
Comment (Vic-Dev) [reply pkg/server/commands/reverseexpand/pipeline.go:442]: that is intended

File: pkg/server/commands/reverseexpand/pipeline.go [hunk 5]
```diff
@@ -411,31 +411,36 @@ func (r *baseResolver) process(ndx int, snd *sender, listeners []*listener) loop
...
-			r.mutexes[ndx].Lock()
-			if _, ok := r.inBuffers[ndx][item.Value]; ok {
-				r.mutexes[ndx].Unlock()
-				continue
 			}
-			r.inBuffers[ndx][item.Value] = struct{}{}
-			r.mutexes[ndx].Unlock()
-			unseen = append(unseen, item)
+		} else {
+			unseen = msg.Value.Items
```
Comment (adriantam) [inline pkg/server/commands/reverseexpand/pipeline.go:441]: Do we have any unit test validating this path?

Comment (Vic-Dev) [reply pkg/server/commands/reverseexpand/pipeline.go:441]: Added

**Comment by adriantam** [comment]
@claude

File: pkg/server/commands/reverseexpand/pipeline.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (Vic-Dev) [reply pkg/server/commands/reverseexpand/pipeline.go:441]: Yes that is the point

File: pkg/server/commands/reverseexpand/pipeline.go [hunk 6]
```diff
@@ -794,29 +797,13 @@ func (r *intersectionResolver) resolve(senders []*sender, listeners []*listener)
...
-				// If there are no unseen items, skip processing
-				if len(unseen) == 0 {
+				// If there are no items, skip processing
+				if len(msg.Value.Items) == 0 {
 					msg.done()
 					return true
 				}
 
-				results := r.interpreter.interpret(ctx, snd.edge(), unseen)
+				results := r.interpreter.interpret(ctx, snd.edge(), msg.Value.Items)
```
[the remark this answers is not in the document]
Comment (Vic-Dev) [reply pkg/server/commands/reverseexpand/pipeline.go:807]: yes that was the point of the PR as well
